frescobaldi_app/scorewiz/parts/_base.py

```diff
@@ -163,6 +167,14 @@ def translateMidiInstrumentWidgets(self):
     def buildStaff(self, data, builder, name, octave, numVoices=1, node=None, clef=None):
         """Build a staff with the given number of voices and name."""
         staff = ly.dom.Staff(name, parent=node)
+        if (self.dynamicsStaff.isChecked()
+            and self.upperVoices.value() and self.lowerVoices.value()):
+            # both staffs have to be present to use this feature
+            w = staff.getWith()
+            # This is a workaround to allow non-printing dynamics for MIDI
+            # output, which is not affected by the Dynamics staff type.
```

Comment:
Hmm, and that produces the intended output? I've never had good results mixing dynamics and `<< >>`.

Reply:
Huh, I learned something new today. Since we're on the subject, is there an elegant way to use `<< >>` within the `up` or `down` parts and still get correct dynamics? The only way I've gotten that to work is to hard-code the dynamics in each voice (an even uglier mess). I know the Score Wizard can create multiple voices per hand, but for the music I write -- pretentious rock, not J.S. Bach -- it's easier to default to one voice and split it with `<< >>` as needed.

Reply:
I've removed the `\omit` lines and MIDI warning from the part generation, so now the only thing that option does is creates the `Dynamics` staff. After thinking about it some more, I'm not completely satisfied with either of our approaches, since it seems like both are ultimately complicated ways to work around LilyPond's leaky abstractions. Unless you have a better idea, I say leave it to the users to pick their choice of nightmares.

Unrelatedly, I also threw in some more fixes for the guitar and contrabass types.